client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -0,0 +1,364 @@
+     * {@link GroupProperties#properties} describes generic properties for a group.
+     * 
+     * @param groupId Id for the group.
+     * @return Group properties which includes property like Serialization format and compatibility policy. 
+     * @throws ResourceNotFoundException if group is not found.
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    GroupProperties getGroupProperties(String groupId) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Update group's schema validation policy. If previous rules are not supplied, then the update to the rules will be
+     * performed unconditionally. However, if previous rules are supplied, then the update will be performed if and only if
+     * existing {@link GroupProperties#schemaValidationRules} match previous rules. 
+     * 
+     * @param groupId Id for the group. 
+     * @param validationRules New Schema validation rules for the group.
+     * @param previousRules Previous schema validation rules.
+     * @return true if the update was accepted by the service, false if it was rejected because of precondition failure.
+     * Precondition failure can occur if previous rules were specified and they do not match the rules set on the group. 
+     * @throws ResourceNotFoundException if group is not found.
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    boolean updateSchemaValidationRules(String groupId, SchemaValidationRules validationRules, @Nullable SchemaValidationRules previousRules)
+        throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Gets list of latest schemas for each object types registered under the group. Objects are identified by {@link SchemaInfo#type}.
+     * Schemas are retrieved atomically. So all schemas added before this call will be returned by this call. 
+     *
+     * @param groupId Id for the group. 
+     * @return List of different objects within the group.   
+     * @throws ResourceNotFoundException if group is not found.
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    List<SchemaWithVersion> getSchemas(String groupId) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Registers schema to the group. Schemas are validated against existing schemas in the group that share the same 
+     * {@link SchemaInfo#type}.
+     * If group is configured with {@link GroupProperties#allowMultipleTypes} then multiple schemas with distinct
+     * type {@link SchemaInfo#type} could be registered. 
+     * All schemas with same type are assigned monotonically increasing version numbers. 
+     * Implementation of this method is expected to be idempotent. The behaviour of Add Schema API on the schema registry
+     * service is idempotent. If a schema is already registered, its version info is returned by the service.  
```

Comment:
new version is returned

Reply:
I was attempting to highlight the phrasing on this line: `If a schema is already registered, its version info is returned by the service.` it says nothing about the `else` case. It could say:
"Returns the new version info or if a schema is already registered, its version info is returned by the service"